packages/legacy-credentials/src/vcInterop.ts

```diff
@@ -0,0 +1,180 @@
+  )
+  // salt/nonces must be sorted by statement digest (keys) and base58 encoded
+  const salt = Object.entries(credential.claimNonceMap)
+    .map(([hsh, slt]) => [hexToU8a(hsh), stringToU8a(slt)])
+    .sort((a, b) => u8aCmp(a[0], b[0]))
+    .map((i) => base58Encode(i[1]))
+  return {
+    type: constants.ATTESTATION_PROOF_V1_TYPE,
+    block,
+    commitments,
+    salt,
+  }
+}
+
+/**
+ * Transforms an [[ICredential]] object to conform to the KiltCredentialV1 data model.
+ *
+ * @param input An [[ICredential]] object.
+ * @param options Additional required and optional parameters for producing a VC from an [[ICredential]].
+ * @param options.issuer The issuer of the attestation to this credential (attester).
+ * @param options.timestamp Timestamp of the block referenced by blockHash in milliseconds since January 1, 1970, UTC (UNIX epoch).
+ * @param options.chainGenesisHash Optional: Genesis hash of the chain against which this credential is verifiable. Defaults to the spiritnet genesis hash.
+ * @returns A KiltCredentialV1 with embedded KiltAttestationProofV1 proof.
+ */
+export function vcFromICredential(
+  input: ICredential,
+  {
+    issuer,
+    timestamp,
+    chainGenesisHash,
+  }: Pick<
+    Parameters<typeof KiltCredentialV1.fromInput>[0],
+    'chainGenesisHash' | 'timestamp' | 'issuer'
+  >
+): Omit<Types.KiltCredentialV1, 'proof'> {
+  const {
+    legitimations: legitimationsInput,
+    delegationId,
+    rootHash: claimHash,
+    claim,
+  } = input
+  const { cTypeHash, owner: subject, contents: claims } = claim
+  const cType = CType.hashToId(cTypeHash)
+
+  const legitimations = legitimationsInput.map(({ rootHash: legHash }) =>
+    KiltCredentialV1.idFromRootHash(legHash)
+  )
+
+  const vc = KiltCredentialV1.fromInput({
+    claimHash,
+    subject,
+    claims,
+    chainGenesisHash,
+    cType,
+    issuer,
+    timestamp,
+    legitimations,
+    ...(delegationId && { delegationId }),
+  })
+
+  return vc
+}
+
+type Params = Parameters<typeof vcFromICredential>[1] &
+  Parameters<typeof proofFromICredential>[1]
+
+/**
+ * Transforms an [[ICredential]] object to conform to the KiltCredentialV1 data model.
+ *
+ * @param input An [[ICredential]] object.
+ * @param opts Additional required and optional parameters for producing a VC from an [[ICredential]].
+ * @param opts.issuer The issuer of the attestation to this credential (attester).
+ * @param opts.blockHash Hash of any block at which the credential is verifiable (i.e. Attested and not revoked).
+ * @param opts.timestamp Timestamp of the block referenced by blockHash in milliseconds since January 1, 1970, UTC (UNIX epoch).
+ * @param opts.chainGenesisHash Optional: Genesis hash of the chain against which this credential is verifiable. Defaults to the spiritnet genesis hash.
+ * @returns A KiltCredentialV1 with embedded KiltAttestationProofV1 proof.
+ */
+export function toVc(
+  input: ICredential,
+  { blockHash, issuer, chainGenesisHash, timestamp }: Params
```

Comment:
How is a user supposed to retrieve the additional pieces required to convert the credential to a VC? If there is a method, then perhaps it could be specified in the JSDoc of the function. If there's not, we should think about a way to make it easy for them.

Reply:
this requires an indexer, which we probably should deploy somewhere and default to... 
made a ticket (https://github.com/KILTprotocol/ticket/issues/2896)
I guess we need to defer this feature until the indexer is ready?